jxl/src/util/test.rs

```diff
@@ -196,9 +223,29 @@ pub fn read_pfm(b: &[u8]) -> Result<Vec<Image<f32>>, Error> {
     Ok(res)
 }
 
-pub(crate) use assert_all_almost_eq;
+macro_rules! with_in_out_rows {
+    ($data:expr, |$slice:ident| $body:expr) => {{
```

Comment:
I'm pretty sure you can do a macro that can be used like this:

```
in_out_rows!(name_of_the_variable, u_from_arbitrary, border_pixels, num_channels);

// do stuff with `name_of_the_variable`.
```

something like:

```
macro_rules! in_out_rows {
   ($name: ident, $u: expr, $border: expr, $chan: literal) => {
      // define intermediate variables
      let $name = [...]
   }
}
```

I think this would be easier to read, and names defined in macros don't collide with macro callers, so it shouldn't cause issues.

Also, this macro should very likely be in the render pipeline test utils.

Reply:
(maybe you could also add an optional maximum xsize, defaulting to 4096)